src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt

```diff
@@ -9,36 +9,195 @@ import java.util.logging.Logger
+ * most of the phg_v2 commands
+ *
+ * The user-callable commands in this file are:
+ * 1. retrieveAgcContigs: called to pull data using the "agc getctg" command.
+ *    Expected user input:  dbPath:String, ranges:List<String>
+ *    The user range list should have entries that look like one of the following:
  *     contig@genome:start-end
  *     contig@genome
  *     contig:start-end (this will error if there are more than 1 genome with "contig")
  *     contig1 (this will error if there are more than 2 genome with "contig1" )
- *  The dbPath is the path to the folder that contains the assemblies.agc file.  It is
- *  expected this will be the same folder that contains the tiledb datasets.
+ *    Returns: Map<String, NucSeq> where the key is the contig and the value is a Biokotlin NucSeq
+ *
+ * 2. retrieveAgcGenomes: called to pull data using the "agc getset" command.
+ *   Expected user input:  dbPath:String, genomes:List<String>
+ *   The user genome list should have sample names that are in the AGC file.
+ *   Returns: Map<String, NucSeq> where the key is the genome and the value is a Biokotlin NucSeq
+ *
+ * 3. retrieveAgcData: called to pull data using    the "agc listset" or "agc listctg" command.
+ *   Expected user input:  dbPath:String, agcCmdList:List<String>
+ *   a.  For listset: the user agcCmdList should contain a single item: "listset"
+ *       This command  returns a list of all the genomes in the AGC file.
+ *   b.  For listctg: the user agcCmdList should contain "listctg" as its first entry.
+ *       The remaining entries should be genome names whose contigs will be listed.
+ *       If no genomes names are supplied, an error is thrown
+ *   Result:  a list of strings, one per genome, in the format:
+ *      genomeName:ctg1,ctg2,ctg3,ctg4
+ *      e.g. LineA:1,2
+ *           LineB:1,2
+ *
+ * The rest of the functions in this file are used internally by the above functions.
+ *
  */
 fun retrieveAgcContigs(dbPath:String,ranges:List<String>):Map<String, NucSeq> {
 
-    val commands = buildAgcCommandFromList(dbPath, ranges)
+    val commands = buildAgcCommandFromList(dbPath, "getctg",ranges)
+    val chromNucSeqMap = queryAgc(commands)
+    return chromNucSeqMap
+}
+
+fun retrieveAgcGenomes(dbPath:String,genomes:List<String>):Map<String, NucSeq> {
+
+    val commands = buildAgcCommandFromList(dbPath, "getset",genomes)
     val chromNucSeqMap = queryAgc(commands)
     return chromNucSeqMap
 }
 
+/**
+ * Processes AGC commands listset or listctg
+ */
+fun retrieveAgcData(dbPath:String,agcCmdList:List<String>):List<String>? {
+
+    val agcFile = "${dbPath}/assemblies.agc"
+
+    check(File(agcFile).exists()) { "File assemblies.agc does not exist in folder $dbPath- please send a valid path that indicates the parent folder for your assemblies.agc compressed file." }
+    check(agcCmdList.isNotEmpty()) { "No AGC command was sent to retrieveAgcData" }
+
+    val agcCmd = agcCmdList[0]
```

Comment:
Do we ever use more than the first element?  If not, I would consider using an enum if possible so we can force the commands to be within a controlled vocabulary

Reply:
Yes, we do.  If the command is listctg (vs listset), then  additional elements indicating the genomes needs to be on the list.

listset: returns a list of genomes (sample names) in the agc file.  No additional parameters.
listctg: returns a list of contigs for the specified genomes.  Must have at least 1 genome on the list., so command plus genomes are on the list.